### DISCUSSION TITLE
about duckdb error handle

### DISCUSSION THREAD

**Discussion by xubobbs** [duckdb/duckdb#13951] [General]
hi,
I have a querstion want to get help.
I use duckdb go driver for data source sync from other db to local duckdb database file.
In some time, I occured some duckdb database file exception which caused duckdb query not work forever . such as assert read because of bad storage exception .
I want to do some recover and restart sync if I can catch database db file storage exception.
How can I do ?

**Comment by Tishj** [comment]
> which caused duckdb query not work forever .

That sounds like you hit a Fatal/Internal exception, which results in preventing you from querying the database until it's restarted because we can't guarantee correctness of the internals anymore.

Closing your connection and reopening the database should resolve that issue, ideally you report the cause of those internal/fatal exceptions here so we can fix them because those are not regular acceptable exceptions users should be faced with.
